app/dojo.py

```diff
@@ -0,0 +1,213 @@
+        else:
+            pass
+
+    def allocate_livingspace(self, person):
+        available_livingspace = self.check_room_availability("Living Space")
+        if available_livingspace:
+            person.accomodation = available_livingspace
+            available_livingspace.occupants.append(person)
+            available_livingspace.available_capacity -= 1
+            cprint("\n %s will reside in %s\n" 
+                % (str(person), str(available_livingspace)), "green")
+        elif person.accomodation == "Pending":
+            person.accomodation = "Unallocated"
+            cprint("\n %s has not been allocated to"
+                 "a living space" % (str(person)), "magenta")
+        else:
+            pass
+        
+    def check_room_availability(self, category):
+        try:
+            available_room = choice(
+                    [room for room in self.rooms if room.category == category and room.available_capacity])
+        except IndexError:
+            available_room = False
+        return available_room
+
+    def add_person(self, category, name, surname, accomodation="N"):
+        name = str(name).capitalize() + " " + str(surname).capitalize()
+        accomodation = str(accomodation).capitalize() if accomodation else "N"
+        category = str(category).capitalize()
+        person_is_invalid = self.check_person_invalidity(name, accomodation, category)
+        if person_is_invalid:
+            cprint(person_is_invalid, "magenta")
+            return person_is_invalid
+        else:
+            office = "Pending"
+            accomodation = "Pending" if accomodation == "Y" and category == "Fellow" else ""
+        if category == "Fellow":
+            person = Fellow(name, office, accomodation)
+        else:
+            person = Staff(name, office)
+        self.persons.append(person)
+        cprint("\n %s has been added to the Dojo as a %s" % (name, category), "green")
+        self.allocate_office(person)
+        if accomodation == "Pending":
+            self.allocate_livingspace(person)
+
+    def check_person_invalidity(self, name, accomodation, category):
+        name_split = name.split(" ")        
+        if not name_split[0].isalpha() or not name_split[1].isalpha():
+            return ("\n %s contains non-alphabets\n"
+                "Person not added\n" % name)
+        name_already_exists = name in [
+            person.name for person in self.persons]
+        if name_already_exists:
+            return ("\n %s already exists\n"
+            "Person not added\n" % name)
+        if accomodation not in ("N","Y"):
+            return ("Choose either 'Y' or 'N' for accomodation")
+        if category not in ("Fellow", "Staff"):
+            return "Invalid person category"
+        return False
+
+    def print_room(self, name):
+        name = name.capitalize()
+        room = self.check_if_room_exists(name)
+        if room != "No such room exists":
```

Comment:
In addition, to line 146 you could also have a method for a room that is created but has no occupants.Assuming I created room swift(office) without occupants if call the function to print room swift it should return something like No people in room swift!